Fix wrong id_for_label usage in rendered evaluation edit form

He3lixxx opened the pull request:
Fixes #1769.

Our old code was broken and just happened to work accidentally before django 4: The default widget for a choice field is a [`Select` widget](https://docs.djangoproject.com/en/4.0/ref/forms/widgets/#django.forms.Select). With that, the rendering:
```html
<select id="test" name="test">
   <option>1</option>
   <option>2</option>
</select>
```
wouldn't require any `id_for_label` to be set for the individual values. However, before django 4, this happened to be set, and our code relied on that with no documented guarantee.

This PR changes the button groups to use `RadioSelect` widgets, which would be rendered as:
```html
<input type="radio" id="id_1" name="test" value="1"><label for="id_1">1</label>
<input type="radio" id="id_2" name="test" value="2"><label for="id_2">2</label>
```
i.e. here, every choice gets a seperate ID. This is also guaranteed [by the docs](https://docs.djangoproject.com/en/4.0/ref/forms/widgets/#radioselect):
> To get more granular, you can use each radio button’s tag, choice_label and id_for_label attributes

So this time, if anything changes, we should see an entry in the release notes.

Edit: We can probably make a front end test for this one, it should be detectable just from the rendered html

janno42 commented:
Do you want to add a test? (:

niklasmohrin reviewed (changes requested):
This seems all correct, blocking until test though (as discussed in person)

niklasmohrin commented:
Take this:
```typescript
const formData = await page.evaluate(() => {
    return Object.fromEntries(new FormData(document.getElementById("evaluation-form") as HTMLFormElement));
});
```

He3lixxx commented:
> Take this:
> 
> ```ts
> const formData = await page.evaluate(() => {
>     return Object.fromEntries(new FormData(document.getElementById("evaluation-form") as HTMLFormElement));
> });
> ```

Nice, thanks. Added that, seems to work now.

I've removed the `TomSelectedHtmlSelectElement` interface, because I figured it doesn't really help if we add "This has a .tomselect attribute" to the type system, but then go on to do stuff inside there (-> `.tomselect.options`, access its keys, ...) while only having it declared as `any`. Or, to say it differently: That only hid the `any` behind another interface, but didn't make the code type-safe. I think if we want proper typing, we should just use the proper tomselect types.

---

Codacy complains about magic numbers (in a test, I think they are fine: We _do_ want to click the first button that says "xyz") and `any` / non-null-assertions (Also fine for me: Its a test, I'm fine if it breaks at runtime, I don't want to write additional boilerplate code just to make that explicit.)

niklasmohrin commented on evap/static/ts/tests/frontend/staff-evaluation-edit.ts:
```diff
@@ -0,0 +1,36 @@
+    async page => {
+        const managerId = await page.evaluate(() => {
+            const tomselect = (document.getElementById("id_contributions-0-contributor") as any).tomselect;
+            const options = tomselect!.options;
```
```suggestion
            const options = tomselect.options;
```
